// @rnwonder/simple-datejs@1.0.8/dist/general-D5nVjUqG.js
import { m as u, a as v, j as g, p } from "./format-CMbjATj9.js";
const b = (e, t, n) => {
  const a = u(t, e, 1 - ((n == null ? void 0 : n.weekStartDay) || 0)).getDay(), y = u(t, e + 1, 0).getDate(), o = [], s = e === 0 ? 11 : e - 1, i = u(s === 11 ? t - 1 : t, s + 1, 0).getDate();
  let r = i - a + 1;
  r === i + 1 && (r = 1);
  const h = (m) => p(m, n == null ? void 0 : n.locale);
  for (let m = 0; m < a; m++)
    o.push({ value: h(r + m), month: "prev" });
  for (let m = 1; m <= y; m++)
    o.push({ value: h(m), month: "current" });
  const d = 35 - o.length >= 0 ? 35 - o.length : 42 - o.length;
  for (let m = 1; m <= d; m++)
    o.push({ value: h(m), month: "next" });
  return o;
}, w = () => {
  const e = /* @__PURE__ */ new Date();
  return { day: e.getDate(), month: e.getMonth(), year: e.getFullYear() };
};
function M(e, t) {
  const n = [];
  for (let a = t; a >= e; a--)
    n.push(a);
  return n;
}
const k = (/* @__PURE__ */ new Date()).getFullYear(), l = (e, t) => t === "prev" ? e === 0 ? 11 : e - 1 : t === "next" ? e === 11 ? 0 : e + 1 : e, c = (e, t, n) => n === "prev" ? t === 0 ? e - 1 : e : n === "next" && t === 11 ? e + 1 : e, T = (e, t) => ({ year: t === 0 ? e - 1 : e, month: t === 0 ? 11 : t - 1 }), j = (e, t) => ({ year: t === 11 ? e + 1 : e, month: t === 11 ? 0 : t + 1 });
function L(e, t) {
  const n = [];
  for (let a = 0; a < e.length; a += t)
    n.push(e.slice(a, a + t));
  return n;
}
const E = ({ startDay: e, month: t, year: n, type: a, setStartDay: y }) => {
  if (a === "single") {
    const o = { ...e || w(), ...t !== void 0 && { month: t }, ...n !== void 0 && { year: n } };
    return y == null || y(o), { selectedDate: o, type: a };
  }
  return null;
}, x = () => {
  const e = /* @__PURE__ */ new Date();
  return { day: e.getDate(), month: e.getMonth(), year: e.getFullYear() };
}, q = (e) => {
  const t = x();
  return e instanceof Date ? e.getDate() === t.day && e.getMonth() === t.month && e.getFullYear() === t.year : e.day === t.day && e.month === t.month && e.year === t.year;
}, R = (e, t) => {
  const n = e instanceof Date ? e : v(e), a = t instanceof Date ? t : v(t);
  return n.getTime() < a.getTime();
}, W = (e, t = "long", n) => new Date(2e3, e, 1).toLocaleString(n ?? "en", { month: t }), z = (e, t) => {
  const n = (a) => {
    e && !e.contains(a.target) && t(a);
  };
  return document.addEventListener("click", n), () => {
    document.removeEventListener("click", n);
  };
}, A = ({ day: e, endDate: t, startDate: n, year: a, month: y, monthStatus: o }) => {
  if (!n || !t)
    return !1;
  const s = u(c(a, y, o), l(y, o), e), i = u(n.year, n.month, n.day), r = u(t.year, t.month, t.day);
  return s > i && s < r;
}, B = ({ dateRange: e, day: t, year: n, month: a, monthStatus: y }) => {
  if (!e)
    return !1;
  const o = u(c(n, a, y), l(a, y), t), s = u(e.year, e.month, e.day);
  return o.getTime() === s.getTime();
}, C = ({ maxDate: e, minDate: t, day: n, year: a, month: y }) => {
  if (!t && !e)
    return !1;
  const o = u(c(g(a), g(y), n.month), l(g(y), n.month), n.value);
  if (t && e) {
    const s = u(t.year, t.month, t.day), i = u(e.year, e.month, e.day);
    return o < s || o > i;
  }
  return t ? o < u(t.year, t.month, t.day) : e ? o > u(e.year, e.month, e.day) : !1;
}, S = ({ disabledDays: e, month: t, day: n, year: a }) => !!e && !!e.find((y) => {
  if ("start" in y && "end" in y) {
    const o = v(y.start), s = v(y.end), i = v({ day: n.value, month: l(t, n.month), year: c(a, t, n.month) });
    return i >= o && i <= s;
  }
  return y.day === n.value && y.month === l(t, n.month) && y.year === c(a, t, n.month);
}), F = ({ enabledDays: e, day: t, year: n, month: a, next: y, prev: o }) => !!e && e.every((s) => {
  const i = t ? { day: t.value, month: l(a, t.month), year: c(n, a, t.month) } : { year: n, month: a };
  if ("start" in s && "end" in s) {
    if (function(r, h, d, m) {
      if (m != null && m.next)
        return h.year === void 0 || r.year === void 0 || h.year > r.year || r.year === h.year && (r.month === void 0 || h.month === void 0 || h.month >= r.month);
      if (m != null && m.prev)
        return d.year === void 0 || r.year === void 0 || d.year < r.year || r.year === d.year && (r.month === void 0 || d.month === void 0 || d.month <= r.month);
      const D = h.year === void 0 || r.year === void 0 || r.year > h.year || r.year === h.year && (r.month === void 0 || h.month === void 0 || r.month > h.month || r.month === h.month && (!r.day || h.day === void 0 || r.day >= h.day)), f = d.year === void 0 || r.year === void 0 || r.year < d.year || r.year === d.year && (r.month === void 0 || d.month === void 0 || r.month < d.month || r.month === d.month && (!r.day || d.day === void 0 || r.day <= d.day));
      return D && f;
    }(i, s.start, s.end, { next: y, prev: o }))
      return !1;
  } else if (function(r, h) {
    return r.day ? r.year === h.year && r.month === h.month && r.day === h.day : r.year === h.year && r.month === h.month;
  }(i, s))
    return !1;
  return !0;
}), G = (e, t, n) => {
  for (let a = new Date(e); a <= t; a.setDate(a.getDate() + 1))
    if (S({ day: { month: "current", value: a.getDate() }, month: a.getMonth(), year: a.getFullYear(), disabledDays: n }))
      return !0;
  return !1;
}, H = (e, t, n) => {
  for (let a = new Date(e); a <= t; a.setDate(a.getDate() + 1))
    if (F({ day: { month: "current", value: a.getDate() }, month: a.getMonth(), year: a.getFullYear(), enabledDays: n }))
      return !0;
}, I = ({ year: e, month: t, day: n }) => {
  const a = l(t, n.month), y = c(e, t, n.month), o = u(y, a, n.value).getDay();
  return { isWeekend: o === 0 || o === 6, isSaturday: o === 6, isSunday: o === 0 };
}, J = (e, t) => e.day === t.day && e.month === t.month && e.year === t.year;
export {
  W as a,
  z as b,
  q as c,
  F as d,
  E as e,
  l as f,
  x as g,
  B as h,
  R as i,
  A as j,
  c as k,
  I as l,
  S as m,
  C as n,
  J as o,
  G as p,
  H as q,
  L as r,
  k as s,
  M as t,
  j as u,
  T as v,
  b as w
};
