// @rnwonder/simple-datejs/dist/general-D5nVjUqG.js
import { m as c, p as k, a as l, j as M } from "./format-CMbjATj9.js";
const E = (e, r, n) => {
  const a = c(
    r,
    e,
    1 - ((n == null ? void 0 : n.weekStartDay) || 0)
  ).getDay(), s = c(r, e + 1, 0).getDate(), o = [], i = e === 0 ? 11 : e - 1, y = i === 11 ? r - 1 : r, g = c(
    y,
    i + 1,
    0
  ).getDate();
  let v = g - a + 1;
  v === g + 1 && (v = 1);
  const m = (u) => k(u, n == null ? void 0 : n.locale);
  for (let u = 0; u < a; u++)
    o.push({ value: m(v + u), month: "prev" });
  for (let u = 1; u <= s; u++)
    o.push({ value: m(u), month: "current" });
  const d = 35 - o.length >= 0 ? 35 - o.length : 42 - o.length;
  for (let u = 1; u <= d; u++)
    o.push({ value: m(u), month: "next" });
  return o;
}, w = () => {
  const e = /* @__PURE__ */ new Date();
  return {
    day: e.getDate(),
    month: e.getMonth(),
    year: e.getFullYear()
  };
};
function P(e, r) {
  const n = [];
  for (let t = r; t >= e; t--)
    n.push(t);
  return n;
}
const D = /* @__PURE__ */ new Date(), R = D.getFullYear(), f = (e, r) => r === "prev" ? e === 0 ? 11 : e - 1 : r === "next" ? e === 11 ? 0 : e + 1 : e, h = (e, r, n) => n === "prev" ? r === 0 ? e - 1 : e : n === "next" && r === 11 ? e + 1 : e, W = (e, r) => ({
  year: r === 0 ? e - 1 : e,
  month: r === 0 ? 11 : r - 1
}), b = (e, r) => ({
  year: r === 11 ? e + 1 : e,
  month: r === 11 ? 0 : r + 1
});
function j(e, r) {
  const n = [];
  for (let t = 0; t < e.length; t += r)
    n.push(e.slice(t, t + r));
  return n;
}
const L = ({
  startDay: e,
  month: r,
  year: n,
  type: t,
  setStartDay: a
}) => {
  if (t === "single") {
    const s = {
      ...e || w(),
      ...r !== void 0 && { month: r },
      ...n !== void 0 && { year: n }
    };
    return a == null || a(s), {
      selectedDate: s,
      type: t
    };
  }
  return null;
}, O = () => {
  const e = /* @__PURE__ */ new Date();
  return {
    day: e.getDate(),
    month: e.getMonth(),
    year: e.getFullYear()
  };
}, B = (e) => {
  const r = O();
  return e instanceof Date ? e.getDate() === r.day && e.getMonth() === r.month && e.getFullYear() === r.year : e.day === r.day && e.month === r.month && e.year === r.year;
}, N = (e, r) => {
  const n = e instanceof Date ? e : l(e), t = r instanceof Date ? r : l(r);
  return n.getTime() < t.getTime();
}, q = (e, r = "long", n) => new Date(2e3, e, 1).toLocaleString(n ?? "en", { month: r }), C = (e, r) => {
  const n = (t) => {
    e && !e.contains(t.target) && r(t);
  };
  return document.addEventListener("click", n), () => {
    document.removeEventListener("click", n);
  };
}, x = ({
  day: e,
  endDate: r,
  startDate: n,
  year: t,
  month: a,
  monthStatus: s
}) => {
  if (!n || !r)
    return !1;
  const o = c(
    h(t, a, s),
    f(a, s),
    e
  ), i = c(n.year, n.month, n.day), y = c(r.year, r.month, r.day);
  return o > i && o < y;
}, V = ({
  dateRange: e,
  day: r,
  year: n,
  month: t,
  monthStatus: a
}) => {
  if (!e)
    return !1;
  const s = c(
    h(n, t, a),
    f(t, a),
    r
  ), o = c(e.year, e.month, e.day);
  return s.getTime() === o.getTime();
}, $ = ({
  maxDate: e,
  minDate: r,
  day: n,
  year: t,
  month: a
}) => {
  if (!r && !e)
    return !1;
  const s = c(
    h(
      M(t),
      M(a),
      n.month
    ),
    f(M(a), n.month),
    n.value
  );
  if (r && e) {
    const o = c(r.year, r.month, r.day), i = c(e.year, e.month, e.day);
    return s < o || s > i;
  } else if (r) {
    const o = c(r.year, r.month, r.day);
    return s < o;
  } else if (e) {
    const o = c(e.year, e.month, e.day);
    return s > o;
  }
  return !1;
}, Y = ({
  disabledDays: e,
  month: r,
  day: n,
  year: t
}) => e ? !!e.find((s) => {
  if ("start" in s && "end" in s) {
    const o = l(s.start), i = l(s.end), y = l({
      day: n.value,
      month: f(r, n.month),
      year: h(t, r, n.month)
    });
    return y >= o && y <= i;
  } else
    return s.day === n.value && s.month === f(r, n.month) && s.year === h(t, r, n.month);
}) : !1, T = ({
  enabledDays: e,
  day: r,
  year: n,
  month: t,
  next: a,
  prev: s
}) => e ? e.every((o) => {
  const i = r ? {
    day: r.value,
    month: f(t, r.month),
    year: h(n, t, r.month)
  } : {
    year: n,
    month: t
  };
  if ("start" in o && "end" in o) {
    const y = o.start, g = o.end;
    if (F(i, y, g, { next: a, prev: s }))
      return !1;
  } else if (A(i, o))
    return !1;
  return !0;
}) : !1;
function A(e, r) {
  return e.day ? e.year === r.year && e.month === r.month && e.day === r.day : e.year === r.year && e.month === r.month;
}
function F(e, r, n, t) {
  if (t != null && t.next)
    return r.year === void 0 || e.year === void 0 || r.year > e.year || e.year === r.year && (e.month === void 0 || r.month === void 0 || r.month >= e.month);
  if (t != null && t.prev)
    return n.year === void 0 || e.year === void 0 || n.year < e.year || e.year === n.year && (e.month === void 0 || n.month === void 0 || n.month <= e.month);
  const a = r.year === void 0 || e.year === void 0 || e.year > r.year || e.year === r.year && (e.month === void 0 || r.month === void 0 || e.month > r.month || e.month === r.month && (e.day ? r.day === void 0 || e.day >= r.day : !0)), s = n.year === void 0 || e.year === void 0 || e.year < n.year || e.year === n.year && (e.month === void 0 || n.month === void 0 || e.month < n.month || e.month === n.month && (e.day ? n.day === void 0 || e.day <= n.day : !0));
  return a && s;
}
const G = (e, r, n) => {
  for (let t = new Date(e); t <= r; t.setDate(t.getDate() + 1))
    if (Y({
      day: {
        month: "current",
        value: t.getDate()
      },
      month: t.getMonth(),
      year: t.getFullYear(),
      disabledDays: n
    }))
      return !0;
  return !1;
}, H = (e, r, n) => {
  for (let t = new Date(e); t <= r; t.setDate(t.getDate() + 1))
    if (T({
      day: {
        month: "current",
        value: t.getDate()
      },
      month: t.getMonth(),
      year: t.getFullYear(),
      enabledDays: n
    }))
      return !0;
}, J = ({
  year: e,
  month: r,
  day: n
}) => {
  const t = f(r, n.month), a = h(e, r, n.month), o = c(a, t, n.value).getDay();
  return {
    isWeekend: o === 0 || o === 6,
    isSaturday: o === 6,
    isSunday: o === 0
  };
}, K = (e, r) => e.day === r.day && e.month === r.month && e.year === r.year;
export {
  q as a,
  C as b,
  B as c,
  T as d,
  L as e,
  f,
  O as g,
  V as h,
  N as i,
  x as j,
  h as k,
  J as l,
  Y as m,
  $ as n,
  K as o,
  G as p,
  H as q,
  j as r,
  R as s,
  P as t,
  b as u,
  W as v,
  E as w
};
