// @rnwonder/simple-datejs/dist/libExport-e_IixH4m.js
import { g as k, c as v, i as y, a as B, b as O } from "./general-D5nVjUqG.js";
import { a as $, c as P, f as j, b as A, d as C, e as I, h as S, l as F, i as L } from "./format-CMbjATj9.js";
const M = ({
  inputRef: h,
  dropDownRef: g,
  positionX: H,
  positionY: e
}) => {
  const c = typeof h == "function" ? h() : h, s = typeof g == "function" ? g() : g;
  if (!c || !s)
    return {};
  const o = c == null ? void 0 : c.getBoundingClientRect(), t = s == null ? void 0 : s.offsetHeight, r = s == null ? void 0 : s.offsetWidth, l = window.innerHeight - (o == null ? void 0 : o.bottom), i = o == null ? void 0 : o.top, D = o == null ? void 0 : o.left, W = window.innerWidth - (o == null ? void 0 : o.right), d = window.innerWidth;
  let a, T;
  const b = e === "top" || e === "bottom", w = `${(o == null ? void 0 : o.top) - t - 10}px`, x = `${o == null ? void 0 : o.bottom}px`;
  e === "top" && (a = w), e === "bottom" && (a = x), l > t && !b ? a = x : (i > t && !b || l < t && e === "bottom" && i > t) && (a = w), (!(l > t) && !(i > t) && !b || e === "top" && !(i > t) || e === "bottom" && !(l > t) && !(i > t)) && (a = "0px");
  let f = 0;
  if (H === "left")
    f = D >= r ? D - r : 0;
  else if (H === "right")
    f = W >= r ? o == null ? void 0 : o.right : d - r;
  else {
    const m = (o == null ? void 0 : o.left) + (o == null ? void 0 : o.width) / 2, n = r / 2;
    m - n >= 0 && m + n <= d ? f = m - n : m - n < 0 ? f = 0 : f = d - r;
  }
  return T = `${f}px`, { top: a, left: T };
}, q = () => ({
  getToday: k,
  convertDateObjectToDate: $,
  convertDateToDateObject: P,
  checkIfItsTodayDate: v,
  isBeforeDate: y,
  getMonthName: B,
  formatDate: j,
  clickOutside: O,
  convert24HourTo12Hour: A,
  getAmPm: C,
  convert12HourTo24Hour: I,
  getCurrentTime: S,
  smartDropDownPosition: M,
  labelFormat: F,
  leadingZeros: L
});
export {
  q as e
};
